src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/IsoSpec.cpp

```diff
@@ -124,9 +114,8 @@ namespace OpenMS
       IP[i] = isotopeProbabilities[i].data();
     }
 
-    Iso* iso = new Iso(dimNumber, isotopeNr.data(), atomCounts.data(), IM, IP);
-    run_(iso);
-    delete iso;
+    Iso iso(dimNumber, isotopeNr.data(), atomCounts.data(), IM, IP);
+    return run_(iso);
```

Comment:
this actually leaks memory (IM and IP are not deleted)

Reply:
maybe you can change that into `std::vector<double*>` IM, IP;` which would fix the memory leak?

Reply:
*facepalm*

of course. Silly mistake. Strange though, I would have thought the compiler should be smart enough to warn about there being leftover code after the return statement... Oh well.

Actually, all we have to do is put the deletes before the return statement, and that's it. Long-term, yeah, adding methods that accept vectors in place of double** here is definitely landing in our TODO list - we're planning to finalize a 1.9.0 release in a few days, clean up the old/experimental cruft that's scattered around our codebase, and then we'll do a code dump on you ;) and fix this too.

Reply:
Fixed by commit ab8b6133ab02551d02325f5498c03402d24fc6f9